Unit tests break with go 1.11

Unit tests break with go version 1.11.
End to end tests are not affected.

## Expected Behavior

All unit tests pass, just like with go 1.10.

## Current Behavior

Tests fail.

## Possible Solution

- The build script might have to regenerate some auto-generated code?
- mitigation/early detection: have CI run tests against go nightly/dev/tip (whatever they call it) once a day

## Steps to Reproduce (for bugs)

1. Setup your environment to use go 1.11
2. `./build.sh ci`

## Context

```
$ GOCACHE=off ./build.sh ci
Running tool & plugin builds...
Building tools/cat for linux-amd64
Building tools/false for linux-amd64
Building tools/sleep for linux-amd64
Building tools/true for linux-amd64
Building handlers/slack for linux-amd64
Running unit tests...
# github.com/sensu/sensu-go/backend/monitor
backend/monitor/supervisor.go:105: Entry.Info call has possible formatting directive %s
--- FAIL: TestHandleTokenSubstitution (0.04s)
	Error Trace:	check_handler_internal_test.go:178
	Error:      	Not equal: 
	            	expected: int32(0)
	            	actual: uint32(0x3)
	Error Trace:	check_handler_internal_test.go:179
	Error:      	"unmatched token: template: :1:42: executing "" at <.Missing>: map has no entry for key "Missing"" does not contain "TestTokenSubstitution devops defaultValue"
--- FAIL: TestTokenSubstitution (0.01s)
    --- FAIL: TestTokenSubstitution/default_value_for_missing_field (0.00s)
        token_test.go:103: unexpected error: unmatched token: template: :1:21: executing "" at <.Check.Foo>: map has no entry for key "Foo"
    --- FAIL: TestTokenSubstitution/default_int_value_for_missing_field (0.00s)
        token_test.go:103: unexpected error: unmatched token: template: :1:21: executing "" at <.Check.Foo>: map has no entry for key "Foo"
FAIL
FAIL	github.com/sensu/sensu-go/agent	2.664s
ok  	github.com/sensu/sensu-go/agent/assetmanager	1.085s
ok  	github.com/sensu/sensu-go/agent/transformers	1.038s
ok  	github.com/sensu/sensu-go/asset	1.031s
ok  	github.com/sensu/sensu-go/backend	1.082s [no tests to run]
ok  	github.com/sensu/sensu-go/backend/agentd	1.062s
?   	github.com/sensu/sensu-go/backend/apid	[no test files]
ok  	github.com/sensu/sensu-go/backend/apid/actions	1.246s
ok  	github.com/sensu/sensu-go/backend/apid/graphql	1.101s
ok  	github.com/sensu/sensu-go/backend/apid/graphql/globalid	1.048s
ok  	github.com/sensu/sensu-go/backend/apid/graphql/relay	1.028s
?   	github.com/sensu/sensu-go/backend/apid/graphql/schema	[no test files]
ok  	github.com/sensu/sensu-go/backend/apid/middlewares	2.226s
ok  	github.com/sensu/sensu-go/backend/apid/routers	1.469s
ok  	github.com/sensu/sensu-go/backend/authentication/jwt	1.019s
ok  	github.com/sensu/sensu-go/backend/authorization	1.051s
?   	github.com/sensu/sensu-go/backend/cmd	[no test files]
?   	github.com/sensu/sensu-go/backend/daemon	[no test files]
ok  	github.com/sensu/sensu-go/backend/dashboardd	1.032s
ok  	github.com/sensu/sensu-go/backend/etcd	1.024s
ok  	github.com/sensu/sensu-go/backend/eventd	1.058s
ok  	github.com/sensu/sensu-go/backend/keepalived	1.040s
ok  	github.com/sensu/sensu-go/backend/messaging	1.030s
?   	github.com/sensu/sensu-go/backend/migration	[no test files]
ok  	github.com/sensu/sensu-go/backend/pipelined	3.300s
ok  	github.com/sensu/sensu-go/backend/schedulerd	1.135s
ok  	github.com/sensu/sensu-go/backend/seeds	5.581s
ok  	github.com/sensu/sensu-go/backend/store	1.016s
ok  	github.com/sensu/sensu-go/backend/store/etcd	1.035s [no tests to run]
ok  	github.com/sensu/sensu-go/cli	1.017s
ok  	github.com/sensu/sensu-go/cli/client	1.026s
ok  	github.com/sensu/sensu-go/cli/client/config/basic	1.021s
ok  	github.com/sensu/sensu-go/cli/commands	1.063s
ok  	github.com/sensu/sensu-go/cli/commands/asset	1.044s
ok  	github.com/sensu/sensu-go/cli/commands/check	1.054s
ok  	github.com/sensu/sensu-go/cli/commands/check/subcommands	1.183s
ok  	github.com/sensu/sensu-go/cli/commands/cluster	1.027s
ok  	github.com/sensu/sensu-go/cli/commands/completion	1.023s
ok  	github.com/sensu/sensu-go/cli/commands/config	1.031s
ok  	github.com/sensu/sensu-go/cli/commands/configure	1.017s
ok  	github.com/sensu/sensu-go/cli/commands/create	1.059s
ok  	github.com/sensu/sensu-go/cli/commands/entity	1.161s
ok  	github.com/sensu/sensu-go/cli/commands/environment	1.063s
ok  	github.com/sensu/sensu-go/cli/commands/event	1.079s
ok  	github.com/sensu/sensu-go/cli/commands/extension	1.032s
ok  	github.com/sensu/sensu-go/cli/commands/filter	1.038s
ok  	github.com/sensu/sensu-go/cli/commands/filter/subcommands	1.023s
ok  	github.com/sensu/sensu-go/cli/commands/handler	1.032s
ok  	github.com/sensu/sensu-go/cli/commands/helpers	1.040s
ok  	github.com/sensu/sensu-go/cli/commands/hook	1.064s
ok  	github.com/sensu/sensu-go/cli/commands/logout	1.027s
ok  	github.com/sensu/sensu-go/cli/commands/mutator	1.057s
ok  	github.com/sensu/sensu-go/cli/commands/organization	1.032s
ok  	github.com/sensu/sensu-go/cli/commands/role	1.046s
ok  	github.com/sensu/sensu-go/cli/commands/root	1.019s
ok  	github.com/sensu/sensu-go/cli/commands/silenced	1.053s
ok  	github.com/sensu/sensu-go/cli/commands/timeutil	1.018s
ok  	github.com/sensu/sensu-go/cli/commands/user	1.058s
ok  	github.com/sensu/sensu-go/cli/commands/version	1.018s
ok  	github.com/sensu/sensu-go/cli/elements/globals	1.026s
ok  	github.com/sensu/sensu-go/cli/elements/list	1.014s
ok  	github.com/sensu/sensu-go/cli/elements/report	1.015s
ok  	github.com/sensu/sensu-go/cli/elements/table	1.017s
ok  	github.com/sensu/sensu-go/command	6.113s
ok  	github.com/sensu/sensu-go/graphql/generator	1.015s
ok  	github.com/sensu/sensu-go/graphql/integration	1.282s
ok  	github.com/sensu/sensu-go/handler	1.018s
ok  	github.com/sensu/sensu-go/handlers/slack	1.056s
ok  	github.com/sensu/sensu-go/rpc	1.042s
ok  	github.com/sensu/sensu-go/system	1.036s
ok  	github.com/sensu/sensu-go/transport	1.014s
ok  	github.com/sensu/sensu-go/types	1.288s
ok  	github.com/sensu/sensu-go/types/dynamic	1.041s
ok  	github.com/sensu/sensu-go/types/v1	1.012s
ok  	github.com/sensu/sensu-go/util/bytes	1.013s
ok  	github.com/sensu/sensu-go/util/eval	1.012s
ok  	github.com/sensu/sensu-go/util/retry	1.110s
ok  	github.com/sensu/sensu-go/util/strings	1.008s
ok  	github.com/sensu/sensu-go/util/url	1.008s
ok  	github.com/sensu/sensu-go/version	1.016s
```

## Your Environment

```
$ go version
go version go1.11 linux/amd64
```
```
$ go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/cyril/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/cyril/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/lib/go"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build313664404=/tmp/go-build -gno-record-gcc-switches"
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -74,6 +74,7 @@ were being displayed rather than the check's current values.
 - Use the provided etcd client TLS information when the flag `--no-embed-etcd` 
 is used.
 - Increase duration delta in TestPeriodicKeepalive integration test.
+- Fixed some problems introduced by Go 1.11.
 
 ### Breaking Changes
 - Removed the KeepaliveTimeout attribute from entities.

--- a/agent/token.go
+++ b/agent/token.go
@@ -79,6 +79,9 @@ func defaultFunc(v ...interface{}) interface{} {
 	if len(v) == 1 {
 		return v[0]
 	} else if len(v) == 2 {
+		if v[1] == nil {
+			return v[0]
+		}
 		return v[1]
 	}
 	return nil

--- a/backend/monitor/supervisor.go
+++ b/backend/monitor/supervisor.go
@@ -106,7 +106,7 @@ func (m *EtcdSupervisor) Monitor(ctx context.Context, name string, event *types.
 	}
 
 	shutdownFunc := func() {
-		logger.Info("shutting down monitor for %s", key)
+		logger.Infof("shutting down monitor for %s", key)
 	}
 
 	// start the watcher
@@ -172,7 +172,7 @@ func watchMon(ctx context.Context, cli *clientv3.Client, mon *monitor, failureHa
 		)
 
 		if _, err := cli.Lease.Revoke(ctx, mon.leaseID); err != nil {
-			logger.WithError(err).Warningf("could not revoke the lease %s for the key %s",
+			logger.WithError(err).Warningf("could not revoke the lease %v for the key %s",
 				mon.leaseID, mon.key,
 			)
 		}

--- a/backend/queue/queue_test.go
+++ b/backend/queue/queue_test.go
@@ -235,7 +235,6 @@ func TestDequeueParallel(t *testing.T) {
 	// If we had multiple errors, only the last one is saved
 	require.NoError(t, errEnqueue)
 	results := make(map[string]struct{})
-	var errDequeue error
 	wg.Add(len(items))
 	for range items {
 		go func() {
@@ -245,7 +244,6 @@ func TestDequeueParallel(t *testing.T) {
 			mu.Lock()
 			defer mu.Unlock()
 			if err != nil {
-				errDequeue = err
 				return
 			}
 			results[item.Value()] = struct{}{}

